src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -49,6 +49,29 @@ class NativeInterface
         private bool Unity2019OrNewer;
         private Thread MainThread;
 
+        private class OnSessionCallback : AndroidJavaProxy
+        {
+
+            private Configuration _config;
+
+            public OnSessionCallback(Configuration config) : base("com.bugsnag.android.OnSessionCallback")
+            {
+                _config = config;
+            }
+            public bool onSession(AndroidJavaObject session)
+            {
+                var wrapper = new NativeSession(session);
+                foreach (var callback in _config.GetOnSessionCallbacks())
+                {
+                    if (!callback.Invoke(wrapper))
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            }
```

Comment:
Implementing a Java interface with a C# class is pure magic

Reply:
🧙